File: c/3151.c
bool isArraySpecial(int* nums, int n) {
    int p = (nums[0] & 1) ^ 1;
    for (int i = 0; i < n; i++) {
        if ((nums[i] & 1) == p)
            return false;
        p ^= 1;
    }
    return true;
}

File: javascript/3151.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isArraySpecial = function (nums) {
    let p = (nums[0] & 1) ^ 1;
    for (let i of nums) {
        if ((i & 1) == p)
            return false;
        p ^= 1;
    }
    return true;
};
